lib/cartopy/mpl/geoaxes.py

```diff
@@ -280,6 +281,31 @@ def set_position(self, position):
             'GeoSpine does not support changing its position.')
 
 
+def _add_transform(func):
+    """A wrapper that adds and validates the transform kwarg."""
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        transform = kwargs.get('transform', None)
+        if transform is None:
+            transform = self.projection
+        if (isinstance(transform, ccrs.CRS) and
+                not isinstance(transform, ccrs.Projection)):
+            raise ValueError('Invalid transform:'
+                             ' Spherical transforms are not supported - '
+                             ' consider using PlateCarree/RotatedPole.')
```

Comment:
This error seems a bit weird now; there are many `Projection`s other than these two which appear spherical-ish.

Also, you should be able to insert the name of the method by inspecting `func`.

Reply:
I agree. I was really just trying to maintain consistency with everything else. This could all be tidied up in the future. And it will hopefully be easier with a single source that needs updating rather than each function.